gulpfile.js

```diff
@@ -198,10 +198,10 @@ async function updateBuildNumber(args) {
             versionParts.length > 2 ? versionParts[2].replace(/(\d+)/, args.buildNumber) : args.buildNumber;
         const newVersion =
             versionParts.length > 1
-                ? `${versionParts[0]}.${versionParts[1]}.${versionParts[0].substring(
+                ? `${versionParts[0]}.${versionParts[1]}.${versionParts[2].substring(
                       0,
                       3
-                  )}${buildNumberPortion.substring(3)}`
+                  )}${buildNumberPortion.substring(0, buildNumberPortion.length - 3)}`
```

Comment:
This build number is generated from the GITHUB_RUN_ID--is it possible that this generates collisions (i.e. the last 3 digits of the ID are incremented but the first n-3 are not)? 

https://github.com/microsoft/vscode-jupyter/blob/b83bac0a6101797a45c393163bb1c546c823b5bb/.github/actions/build-vsix/action.yml#L15

Reply:
> ble that this generates collisions (i.e. the last 3 digits of the ID are incremented between runs but the first n-3 are not)?

Not for biulds that happen every night/daily.